### PR TITLE
Huijie - fix ability to edit start and end date

### PR DESCRIPTION
**Pull request by metaphor987**
# Description
task description
Video Link:
https://www.loom.com/share/331a47ec61094572ad7a8665c76059a3?sid=8b7adba3-784f-478f-b977-c6aa3741e24c

## Related PRS (if any):
This frontend PR is related to the #1092 backend PR (https://github.com/OneCommunityGlobal/HGNRest/pull/1092). 

## Main changes explained:
1. Remove the error message that prevents users from editing the start and end dates.
2. If the account created date is before 2022-01-01, then the start date could be set to a date before the account created date. Otherwise, it can only be set to a date on or after the created date. When the start date doesn't meet the requirement, a form validation message will be displayed and the save button will be disabled.

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. clear site data/cache
4. log as an owner or admin user
5. either change the user's own dates, or change another user's dates
To change dates for themselves: go to dashboard → View Profile → Volunteering Time
To change another user's dates: go to Other Link → User Management

6. Make sure the account whose date will be modified have a jobTitle! If the account has the jobTitle, you can find it in the userProfile page right under the user's name. One way to add the jobTitle for the user is to directly edit it in the database. If the jobTitle field is empty, there will be a mongoose validation error when saving the updated dates.

7. verify that (1) when the account created date is before 2022-01-01, an admin/owner can modify the user's start date to a date before the created date. (2) when the account created date is after 2022-01-01, the admin/owner can only modify the user's start date to a date after the created date.

## Screenshots or videos of changes:
The userprofile page will act as follows:
1. For accounts with created date before 2022
https://www.dropbox.com/scl/fi/o3e7h22x95oeq14wkobmz/created-date-before.mov?rlkey=krhv8mvpctmcbskgrwe5p5k9f&st=4ops17c2&dl=0
2. For accounts with created date after 2022
https://www.dropbox.com/scl/fi/ujtinm19d3dt9tybxuam2/created-date-after.mov?rlkey=lchwsvxn3aszvg795lik9lbd7&st=8k1y71gh&dl=0

Thank you!

### CODE REVIEW

**Review by AJAYINAVOLU — APPROVED** [review]
Changes are working as expected.

image

Thank You

**Review by RishithaMamidala — APPROVED** [review]
@metaphor987 
I have reviewed the PR. It works correctly for dates after 2022.
PR 2655

**Review by vishnupriyaatheti — APPROVED** [review]
@metaphor987 I have verified your changes and PR is working as expected.
FE PR 2655+ BE PR 1092

**Review by manikittu810 — COMMENTED** [review]
Hello @metaphor987 
I was trying to review the PR as you mentioned. Is this the expected behavior?
Screenshot 2024-09-12 at 1 11 02 PM


Thank you.

File: src/components/UserProfile/VolunteeringTimeTab/VolunteeringTimeTab.jsx
Comment (ArtemisNyx3) [inline src/components/UserProfile/VolunteeringTimeTab/VolunteeringTimeTab.jsx:1]: Looks good to me!

**Review by ArtemisNyx3 — APPROVED** [review]
Functionality works 
Code looks good

**Comment by sam19981** [comment]
Able to edit start date and end date.

Screenshot 2024-09-13 at 11 03 56 PM

**Review by sam19981 — APPROVED** [review]

**Review by Gauravmg6 — APPROVED** [review]
The changes are working as expected.
Screenshot 2024-09-16 135402

**Review by Watchdog00 — APPROVED** [review]
I have completed the testing and got the expected results. Good job!

**Review by anirudhsk2107 — APPROVED** [review]
The changes are working as expected.
Screen Shot 2024-09-20 at 5 28 33 PM
Screen Shot 2024-09-20 at 5 28 23 PM
Screen Shot 2024-09-20 at 5 28 08 PM

**Review by Lexie5212 — APPROVED** [review]
I tested the PR, all code changes worked as expected.
PR 2655+1092

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**